contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,350 @@
+MODE_ALL, MODE_NETWORKS, MODE_ENTRIES = MODES
+
+PROTOS = ('http', 'https')
+PROTO_HTTP, PROTO_HTTPS = PROTOS
+
+CONFIG_HTTP_LAST_MODIFIED = 'http-last-modified'
+
+
+def _get_cmd_parser():
+    parser = OptionParser(
+        description='Update configuration in DHCP server.',
+    )
+    parser.add_option('-H', '--host', help='Ralph instance host.')
+    parser.add_option('-k', '--key', help='Ralph API key.')
+    parser.add_option(
+        '-m',
+        '--mode',
+        type='choice',
+        choices=MODES,
+        default=MODE_ALL,
+        help='Choose what part of config you want to upgrade. '
+             '[Default: all; Options: {}]'.format(', '.join(MODES)),
+    )
+    parser.add_option(
+        '-l',
+        '--log-path',
+        help='Path to log file. [Default: STDOUT]',
+        default='STDOUT',
+    )
+    parser.add_option(
+        '-c',
+        '--dhcp-config-entries',
+        help='Path to the DHCP entries configuration file.',
+    )
+    parser.add_option(
+        '-n',
+        '--dhcp-config-networks',
+        help='Path to the DHCP networks configuration file.',
+    )
+    parser.add_option(
+        '-a',
+        '--proto',
+        type='choice',
+        choices=PROTOS,
+        default=PROTO_HTTPS
+    )
+    parser.add_option(
+        '-e',
+        '--net-env',
+        help='Only get config for the specified environment.',
+    )
+    parser.add_option(
+        '-d',
+        '--dc',
+        help='Only get config for the specified data center.',
+    )
+    parser.add_option(
+        '-r',
+        '--restart',
+        help='Restart service after fetch config?',
+        action='store_true',
+    )
+    parser.add_option(
+        '-v',
+        '--verbose',
+        help='Increase verbosity.',
+        action='store_true',
+        default=False,
+    )
+    parser.add_option(
+        '-s',
+        '--dhcp-service-name',
+        help='Name of the service to restart.',
+        default='isc-dhcp-server'
+    )
+    return parser
+
+
+def _setup_logging(filename, verbose=False):
+    log_size = 20  # MB
+    logger = logging.getLogger(__file__)
+    if verbose:
+        logger.setLevel(logging.INFO)
+    else:
+        logger.setLevel(logging.WARNING)
+    if not filename or filename in ('-', 'STDOUT'):
+        handler = logging.StreamHandler()
+    else:
+        handler = logging_handlers.RotatingFileHandler(
+            filename, maxBytes=(log_size * (1 << 20)), backupCount=5
+        )
+    fmt = logging.Formatter("[%(asctime)-12s.%(msecs)03d] "
+                            "%(levelname)-8s %(filename)s:%(lineno)d  "
+                            "%(message)s", "%Y-%m-%d %H:%M:%S")
+    handler.setFormatter(fmt)
+
+    logger.addHandler(handler)
+    return logger
+
+
+def _remove_apication_lock(lockfile, logger):
+    logger.info('Remove lock')
+    os.unlink(lockfile)
+
+
+def _set_script_lock(logger):
+    lockfile = '/tmp/{}.lock'.format(os.path.split(sys.argv[0])[1])
```

Comment:
No, because we need explicit path to lock file. On the other hand ``/tmp/`` is ugly therefore I will use [tempfile.gettempdir](https://docs.python.org/3.5/library/tempfile.html#tempfile.gettempdir)